release/src/main/scripts/download_github_actions_artifacts.py

```diff
@@ -0,0 +1,314 @@
+  if not github_token:
+    return ask_for_github_token()
+  return github_token
+
+
+def request_url(url, github_token, return_json=True, *args, **kwargs):
+  """Helper function form making requests authorized by GitHub token."""
+  r = requests.get(url, *args, auth=("token", github_token), **kwargs)
+  if return_json:
+    r.raise_for_status()
+    return r.json()
+  return r
+
+
+def safe_get(data, key, url=None):
+  """Gets data by the key with informative Exception in case of non existent key."""
+  if key not in data:
+    message = f'There is missing key: "{key}" in response data: {data}.'
+    if url:
+      message += f" Requested url: {url}"
+    raise ValueError(message)
+  return data.get(key)
+
+
+def get_yes_or_no_answer(question):
+  """Asks yes or no question."""
+  reply = str(input(question + " 'y' or 'n'): ")).lower().strip()
+  if reply == "y":
+    return True
+  elif reply == "n":
+    return False
+  else:
+    return get_yes_or_no_answer("Uhhhh... please enter")
+
+
+def get_build_wheels_workflow_id(repo_url, github_token):
+  """Gets workflow id."""
+  url = GH_API_URL_WORKLOW_FMT.format(repo_url=repo_url)
+  data = request_url(url, github_token)
+  return safe_get(data, "id", url)
+
+
+def get_single_workflow_run_data(run_id, repo_url, github_token):
+  """Gets single workflow run data (github api payload)."""
+  url = GH_API_URL_WORKFLOW_RUN_FMT.format(repo_url=repo_url, run_id=run_id)
+  return request_url(url, github_token)
+
+
+def get_last_run_id(
+    workflow_id, repo_url, release_branch, release_commit, github_token):
+  """
+  Gets id of last run for given workflow, repo, branch and commit.
+  Raises exception when no run found.
+  """
+  url = GH_API_URL_WORKFLOW_RUNS_FMT.format(
+      repo_url=repo_url, workflow_id=workflow_id)
+  data = request_url(
+      url,
+      github_token,
+      params={
+          "event": "push", "branch": release_branch
+      },
+  )
+  runs = safe_get(data, "workflow_runs", url)
+
+  filtered_commit_runs = [
+      r for r in runs if r.get("head_sha", "") == release_commit
+  ]
+
+  if not filtered_commit_runs:
+    workflow_run_web_url = GH_API_URL_WORKFLOW_RUNS_FMT.format(
+        repo_url=repo_url, workflow_id=workflow_id)
+    raise Exception(
+        f"No runs for workflow (branch {release_branch}, commit {release_commit}). Verify at {workflow_run_web_url}"
+    )
+
+  sorted_runs = sorted(
+      filtered_commit_runs,
+      key=lambda w: dateutil.parser.parse(w["created_at"]),
+      reverse=True,
+  )
+  last_run = sorted_runs[0]
+  print(
+      f"Found last run. SHA: {release_commit}, created_at: '{last_run['created_at']}', id: {last_run['id']}"
+  )
+  workflow_run_web_url = GH_WEB_URL_WORKLOW_RUN_FMT.format(
+      repo_url=repo_url, run_id=last_run["id"])
+  print(f"Verify at {workflow_run_web_url}")
+  print(
+      f"Upload to GCS available at: gs://beam-wheels-staging/{release_branch}/{release_commit}-{workflow_id}/"
```

Comment:
How about:
```
GCS location corresponding to artifacts built in this run: 
```
Also at the end of GCS path do we have workflow_id or last_run["id"] ?

Reply:
it should be `last_run["id"]`. Thanks for pointing this out. I fix it and applied your suggestion about message.